modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbTableStorage.java

```diff
@@ -0,0 +1,382 @@
+     * {@link CFType}.
+     *
+     * @return Map with column families names.
+     * @throws StorageException If something went wrong.
+     */
+    private Map<CFType, List<String>> getCfNames() {
+        String absolutePathStr = tablePath.toAbsolutePath().toString();
+
+        List<String> cfNames = new ArrayList<>();
+
+        try (Options opts = new Options()) {
+            List<byte[]> cfNamesBytes = RocksDB.listColumnFamilies(opts, absolutePathStr);
+
+            for (byte[] cfNameBytes : cfNamesBytes)
+                cfNames.add(new String(cfNameBytes, StandardCharsets.UTF_8));
+        }
+        catch (RocksDBException e) {
+            throw new StorageException(
+                "Failed to read list of columnfamilies names for the RocksDB instance located at path " + absolutePathStr,
+                e
+            );
+        }
+
+        return cfNames.stream().collect(groupingBy(this::cfType));
+    }
+
+    /**
+     * Returns list of CF descriptors by their names.
+     *
+     * @param cfGrouped Map from CF type to lists of names.
+     * @return List of CF descriptors.
+     */
+    @NotNull private List<ColumnFamilyDescriptor> convertToCfDescriptors(Map<CFType, List<String>> cfGrouped) {
+        List<ColumnFamilyDescriptor> cfDescriptors = new ArrayList<>();
+
+        Options cfOptions = addToCloseableResources(new Options().setCreateIfMissing(true));
+
+        cfDescriptors.add(new ColumnFamilyDescriptor(
+            CF_META.getBytes(StandardCharsets.UTF_8),
+            addToCloseableResources(new ColumnFamilyOptions(cfOptions))
+        ));
+
+        for (String partitionCfName : cfGrouped.getOrDefault(CFType.PARTITION, List.of())) {
+            cfDescriptors.add(new ColumnFamilyDescriptor(
+                partitionCfName.getBytes(StandardCharsets.UTF_8),
+                new ColumnFamilyOptions()
+            ));
+        }
+
+        NamedListView<? extends TableIndexView> indicesCfgView = tableCfg.value().indices();
+
+        for (String indexCfName : cfGrouped.getOrDefault(CFType.INDEX, List.of())) {
+            String indexName = indexCfName.substring(CF_INDEX_PREFIX.length());
+
+            TableIndexView indexCfgView = indicesCfgView.get(indexName);
+
+            assert indexCfgView != null : "Found index that is absent in configuration: " + indexCfName;
+
+            Comparator<ByteBuffer> indexComparator = indexComparatorFactory.apply(tableCfg.value(), indexName);
+
+            cfDescriptors.add(new ColumnFamilyDescriptor(
+                indexCfName.getBytes(StandardCharsets.UTF_8),
+                new ColumnFamilyOptions()
+                    .setComparator(addToCloseableResources(
+                        new AbstractComparator(addToCloseableResources(new ComparatorOptions())) {
+                            /** {@inheritDoc} */
+                            @Override public String name() {
+                                return "index-comparator";
```

Comment:
Why is a comparator name hardcoded? Is it possible that this name will clash with other comparators?